src/bag/simulation/cache.py

```diff
@@ -210,6 +205,39 @@ async def _create_dut(self, impl_cell: str,
                           dut_params: Mapping[str, Any], extract: Optional[bool] = None,
                           name_prefix: str = '', name_suffix: str = '', flat: bool = False,
                           export_lay: bool = False, em: bool = False) -> Tuple[DesignInstance, Optional[Path], bool]:
+        """Check for cached designs and, if not cached, create a new instance.
+
+        Parameters
+        ----------
+        impl_cell : str
+            Cell name
+        dut_cls : Union[Type[TemplateBase], Type[Module], str]
+            Generator template class (DesignMaster)
+        dut_params : Mapping[str, Any]
+            Generator template parameters
+        extract : Optional[bool]
+            If True, create an extracted netlist
+        name_prefix : str
+            Optional additional prefix for all cell names, for uniquification
+        name_suffix : str
+            Optional additional suffix for all cell names, for uniquification
+        flat : bool
+            If True, generate flat netlist
+        export_lay: bool
+            If True, create OA layout view, then produce export to GDS. Else, write directly to GDS.
```

Comment:
Take extra care while implementing this for the without_OA version, since there you make the GDS first and then optionally import it back for OA.